packages/@aws-cdk/aws-ecs/lib/images/tag-parameter-container-image.ts

```diff
@@ -0,0 +1,48 @@
+import * as ecr from '@aws-cdk/aws-ecr';
+import * as cdk from '@aws-cdk/core';
+import { ContainerDefinition } from '../container-definition';
+import { ContainerImage, ContainerImageConfig } from '../container-image';
+
+/**
+ * A special type of {@link ContainerImage} that uses an ECR repository for the image,
+ * but a CloudFormation Parameter for the tag of the image in that repository.
+ * This allows providing this tag through the Parameter at deploy time,
+ * for example in a CodePipeline that pushes a new tag of the image to the repository during a build step,
+ * and then provides that new tag through the Parameter in the deploy step.
+ *
+ * @see #tagParameterName
+ */
+export class TagParameterContainerImage extends ContainerImage {
+  private readonly repository: ecr.IRepository;
+  private imageTagParameter?: cdk.CfnParameter;
+
+  public constructor(repository: ecr.IRepository) {
+    super();
+    this.repository = repository;
+  }
+
+  public bind(scope: cdk.Construct, containerDefinition: ContainerDefinition): ContainerImageConfig {
+    this.repository.grantPull(containerDefinition.taskDefinition.obtainExecutionRole());
+    const imageTagParameter = new cdk.CfnParameter(scope, 'ImageTagParam');
+    this.imageTagParameter = imageTagParameter;
+    return {
+      imageName: this.repository.repositoryUriForTag(imageTagParameter.valueAsString),
```

Comment:
So, the line you linked to grants the CodeBuild Project that performs the `docker push` in the Pipeline permissions to perform that operation. Obviously, the Project knows it needs those permissions, because its Buildspec contains that `docker push` command.

Separately, we allow the Role of the TaskDefinition permissions to pull that image (that's 4 lines above this one, line 25).

Is that enough, or do you think we need more guidance here?

Reply:
Ah, that makes sense. I just wasn't sure if those somehow populated the credentials field in the ContainerImageConfig, but since the correct permissions are added within the bind call it should be fine!